core/pom.xml

```diff
@@ -629,10 +629,34 @@
           <shadeTestJar>true</shadeTestJar>
           <artifactSet>
             <includes>
+              <include>org.spark-project.spark:unused</include>
```

Comment:
Have you compared the compiled logs of Maven shade? If it is not use `append` mode, we need make sure there is no omission.

Reply:
Yes, the result is expected.

This PR

```
[INFO] --- maven-shade-plugin:3.2.4:shade (default) @ spark-core_2.12 ---
[INFO] Including org.eclipse.jetty:jetty-plus:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-security:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-util:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-server:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-io:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-http:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-continuation:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-servlet:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-proxy:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-client:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-servlets:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including com.google.protobuf:protobuf-java:jar:3.21.9 in the shaded jar.
[INFO] Including org.spark-project.spark:unused:jar:1.0.0 in the shaded jar.
```

Before https://github.com/apache/spark/pull/38779
```
[INFO] --- maven-shade-plugin:3.2.4:shade (default) @ spark-core_2.12 ---
[INFO] Including org.eclipse.jetty:jetty-plus:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-security:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-util:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-server:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-io:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-http:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-continuation:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-servlet:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-proxy:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-client:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.eclipse.jetty:jetty-servlets:jar:9.4.49.v20220914 in the shaded jar.
[INFO] Including org.spark-project.spark:unused:jar:1.0.0 in the shaded jar.
```